### ISSUE TITLE
Different inference values tensorflow vs tflite model

### ISSUE THREAD

**Issue by naolenikhil** [tensorflow/tensorflow#42346]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS High Sierra-10.13.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Not checked
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.3.0-rc2-23-gb36436b087 2.3.0
- Python version: 3.8.5
- Bazel version (if compiling from source): 
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**
Model built using tensorflow and later converter to tflite gives different results for same input. See code provided to reproduce the error.

**Describe the expected behavior**
Must produce same output. 

**Standalone code to reproduce the issue**
```
from tensorflow.keras.models import Model, Sequential, load_model
from tensorflow.keras import layers, Input
import tensorflow.keras.backend as K
import tensorflow as tf
import numpy as np
import imutils
import tflite
import cv2


def get_laplacian_kernel(shape, dtype=None):
    h, w, nr_channels, channel_multiplier  = shape
    assert channel_multiplier == 1
    assert h == w == 3
    laplacian_mask = np.array([[ 0,  1, 0],
                               [ 1, -4, 1],
                               [ 0,  1, 0]])
    mask = np.zeros(shape)
    for ch in range(nr_channels):
        mask[:, :, ch, 0] = laplacian_mask
    laplacian_kernel = K.variable(mask, dtype=dtype)
    return laplacian_kernel


def build_laplacian_model():
    # allow input image to be of any size
    input_shape = (1200, 1800, 3)
    input_layer = Input(shape=input_shape, )
    layer_conv_laplacian = layers.DepthwiseConv2D(kernel_size=3,
                                                  depthwise_initializer=get_laplacian_kernel, 
                                                  depth_multiplier=1,
                                                  bias_initializer='zeros',
                                                  strides=1, padding='same',
                                                  trainable=False
                                                  )(input_layer)
    layer_variance = layers.Lambda(lambda tensor: K.var(tensor), trainable=False)(layer_conv_laplacian)
    laplacian_model = Model(inputs=input_layer, outputs=layer_variance)
    return laplacian_model

blur_model = build_laplacian_model()
print(blur_model.summary())


converter = tf.lite.TFLiteConverter.from_keras_model(blur_model)
fpath = 'model.tflite'
with open(fpath, 'wb') as fw:
    fw.write(converter.convert())
print(f'tflite model fpath: {fpath}')


interpreter = tf.lite.Interpreter(model_path=fpath)

def tflite_inference(model, imdata):
    input_details = model.get_input_details()
    output_details = model.get_output_details()
    model.resize_tensor_input(input_details[0]["index"], imdata.shape)
    model.allocate_tensors()
    model.set_tensor(input_details[0]['index'], imdata)
    model.invoke()
    result = model.get_tensor(output_details[0]['index'])
    return result


# inference using both tensorflow and tflite models
print('\n\nInference using both models')
urls = ['https://images.unsplash.com/photo-1470020337050-543c4e581988',
        'https://images.unsplash.com/photo-1516811108838-030371f93644',
        'https://zipbooks.com/wp-content/uploads/2017/05/royalty-free-images-free-of-charge.jpeg',
       ]

for url in urls:
    imdata = imutils.url_to_image(url)
    imdata = cv2.resize(imdata, dsize=(1800, 1200))
    imdata = np.array([imdata]).astype(np.float32)
    pred0 = blur_model.predict(imdata)
    pred1 = tflite_inference(interpreter, imdata)
    print(f'tensorflow: {pred0} \t tflite: {pred1}')
```

**Other info / logs** 
Output
```
Model: "functional_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         [(None, 1200, 1800, 3)]   0         
_________________________________________________________________
depthwise_conv2d (DepthwiseC (None, 1200, 1800, 3)     30        
_________________________________________________________________
lambda (Lambda)              ()                        0         
=================================================================
Total params: 30
Trainable params: 0
Non-trainable params: 30
_________________________________________________________________
None
WARNING:tensorflow:From /Users/<...>/lib/python3.8/site-packages/tensorflow/python/training/tracking/tracking.py:111: Model.state_updates (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
WARNING:tensorflow:From /Users/<...>/lib/python3.8/site-packages/tensorflow/python/training/tracking/tracking.py:111: Layer.updates (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
INFO:tensorflow:Assets written to: /var/<...>/assets
tflite model fpath: model.tflite


Inference using both models
tensorflow: 981.5494995117188 	 tflite: 976.8895874023438
tensorflow: 289.7860412597656 	 tflite: 285.5138854980469
tensorflow: 57.03384017944336 	 tflite: 56.148902893066406
```

**Comment by jvishnuvardhan** [comment]
@naolenikhil I am able to reproduce the issue. [Here](https://colab.research.google.com/gist/jvishnuvardhan/e0688bf5d2093db84478f70b1623fb29/untitled3.ipynb) is the gist for our reference. Thanks!

**Comment by naolenikhil** [comment]
@jvishnuvardhan - Thanks for creating the gist. 
1. Were you able to identify the root cause of this issue?
2. Does a previous version of tensorflow or tflite solve this issue? 
3. Are you expecting any other information from me?

**Comment by naolenikhil** [comment]
Wanted to check on progress of the ticket. I created [#42694 ](url) . I suspect variance calculation is the main culprit here and DepthwiseConv2D operations are fine.

**Comment by srjoglekar246** [comment]
Hey @thaink, could you take a look at this issue? See [this comment](https://github.com/tensorflow/tensorflow/issues/42694#issuecomment-685172606) for more details on the where our reduction ops might be causing loss of precision.

**Comment by SuryanarayanaY** [comment]
The issue replicated in TF2.11V and still persists.Please refer attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/d9fcb6878517f7c31df6dbbf907499cb/42346_2-11.ipynb).

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.
